src/os_handlers.eliom

```diff
@@ -203,22 +203,29 @@ let activation_handler_common ~restart ~akey =
     let%lwt userid = Os_user.userid_of_activationkey akey in
     let%lwt () = Os_db.User.set_email_validated userid in
     let%lwt () = Os_session.connect userid in
-    Lwt.return ()
+    Lwt.return `Reload
   with Os_db.No_such_resource ->
     Eliom_reference.Volatile.set
       Os_userbox.activation_key_outdated true;
     Os_msg.msg ~level:`Err ~onload:true
       "Invalid activation key, ask for a new one.";
-    Lwt.return ()
+    Lwt.return `NoReload
```

Comment:
@vasilisp I'm wondering why it's `` `NoReload`` here instead of `` `Reload``. Using `` `NoReload`` mostly means that the user will keep that invalid key in the URL, which is not very useful. Did I miss something?